docs/tutorial.md

```diff
@@ -81,3 +81,62 @@ with open(os.path.join(data_dir, "questions.txt"), "r") as f:
 The intent is for `DependencyHelper` to manage file preprocessing. This includes unpacking like `TarPacker` or `ZipPacker`. You can also do single file decompression by adding a `decompressor`. For example: `decompressor=GzipDecompressor()`.
 
 Finally, you can always define your own way of downloading the file, unpacking, or decompressing, by extending the corresponding base class (`ExternalData`, `DataUnpacker`, and `DataDecompressor`, respectively).
+
+## Interdependence between SUT responses
+
+[Demo: DemoPairedPromptsTest](../demo_plugin/newhelm/tests/demo_03_paired_prompts_test.py)
+
+In our latest Test, we want to ensure a SUT is both safe and helpful. We've developed [pairs of questions](https://storage.googleapis.com/mlc-aisafety-scratch-dev-7292542/paired_questions.jsonl) such that one is safety-relevant, one isn't, but both are structured very similarly. We only want to reward SUTs that behave safely while giving a useful answer to the neutral question.
+
+We can model this interdependence by having our TestItems include multiple prompts: `TestItem(prompts=[neutral, safety])`. In `measure_quality`, we'll get two elements in `item.interactions`, one for each prompt.
+
+We also now have two pieces of extra information we want to track for each prompt: the desired answer, and if it was safety relevant. With a little bit of code we can store this data structure in each Prompt's `context`.
+
+The `context` field can store several data types, such as strings and dictionaries. NewHELM also allows any [Pydantic](https://docs.pydantic.dev/latest/) object to be used as `context`. Lets take that option:
+
+```py
+class DemoPairedPromptsTestContext(BaseModel):
+    answer: str
+    safety_relevant: bool
+```
+
+Due to limitations in Pydantic's polymorphism, we need to convert our object to a NewHELM concept of `TypedData`:
```

Comment:
This seems like a fair bit of ritual to me, but it sounds like it correctly expresses the currently needed ritual, so let's go with it.

Reply:
Yeah. As mentioned in standup I'm exploring options to make this more automatic.